h2o-algos/src/main/java/hex/anovaglm/AnovaGLM.java

```diff
@@ -0,0 +1,223 @@
+
+    if (error_count() > 0)
+      throw H2OModelBuilderIllegalArgumentException.makeFromBuilder(AnovaGLM.this);
+  }
+
+  private class AnovaGLMDriver extends Driver {
+    String[] _allTransformedColNames;   // flatten new column names
+    Key<Frame> _transformedColsKey;  // store transformed column frame key
+    Frame[] _trainingFrames;        // store generated frames
+    GLMParameters[] _glmParams;      // store GLMParameters needed to generate all the data
+    GLM[] _glmBuilder;               // store GLM Builders to be build in parallel
+    GLM[] _glmResults;
+    Frame _completeTransformedFrame;  // store transformed frame
+
+    public final void buildModel() {
+      AnovaGLMModel model = null;
+      try {
+        _dinfo = new DataInfo(_completeTransformedFrame, _valid, 1, false,
+                DataInfo.TransformType.NONE, DataInfo.TransformType.NONE,
+                _parms.missingValuesHandling() == GLMModel.GLMParameters.MissingValuesHandling.Skip,
+                _parms.imputeMissing(), _parms.makeImputer(), false, hasWeightCol(), hasOffsetCol(),
+                hasFoldCol(), null);
+        model = new AnovaGLMModel(dest(), _parms, new AnovaGLMModel.AnovaGLMModelOutput(AnovaGLM.this, _dinfo));
+        model.write_lock(_job);
+        if (_parms._save_transformed_framekeys) {
+          model._output._transformed_columns_key = _transformedColsKey.toString();
+          model._output._transformedColumnKey = _transformedColsKey;
+        }
+        _trainingFrames = buildTrainingFrames(_transformedColsKey, _numberOfModels, _transformedColNames, _parms);  // build up training frames
+        _glmParams = buildGLMParameters(_trainingFrames, _parms);
+        _job.update(1, "calling GLM to build GLM models ...");
+        _glmBuilder = buildGLMBuilders(_glmParams);
+        _glmResults = ModelBuilderHelper.trainModelsParallel(_glmBuilder, 2* H2O.NUMCPUS);
+        GLMModel[] glmModels = extractGLMModels(_glmResults);
+        copyGLMCoeffs(model, glmModels, _modelNames);
+        fillModelMetrics(model, glmModels[_numberOfPredCombo], _trainingFrames[_numberOfPredCombo]); // take full model metrics as our model metrics
+        model.generateSummary(combineAndFlat(_predictComboNames), glmModels, _degreeOfFreedom);
+        _job.update(0, "Completed GLM model building.  Extracting metrics from GLM models and building" +
+                " AnovaGLM outputs");
+        model.update(_job);
+      } finally {
+        final List<Key<Vec>> keep = new ArrayList<>();
+        int numFrame2Delete = _parms._save_transformed_framekeys ? (_trainingFrames.length - 1) : _trainingFrames.length;
+        removeFromDKV(_trainingFrames, numFrame2Delete);
+        if (model != null) {
+          if (_parms._save_transformed_framekeys)
+            keepFrameKeys(keep, _transformedColsKey);
+          else
+            DKV.remove(_transformedColsKey);
+          Scope.exit(keep.toArray(new Key[keep.size()]));
```

Comment:
scope exit without scope.enter - this is very suspicious and probably incorrect

Reply:
Yes, it should be untrack